pkg/querier/queryrange/split_by_interval.go

```diff
@@ -263,8 +271,42 @@ func forInterval(interval time.Duration, start, end time.Time, callback func(sta
 	}
 }
 
-func splitMetricByTime(r queryrangebase.Request, interval time.Duration) []queryrangebase.Request {
+// maxRangeVector returns the maximum range vector duration within a LogQL query.
+func maxRangeVector(q string) (time.Duration, error) {
```

Comment:
I was a bit confused by the name of this function, since it returns a duration and not a range vector. The function doc string describes the return type correctly, though.

Reply:
I can rename it to `maxRangeVectorDuration` ?

Reply:
doing it.